packages/text/src/vars/index.ts

```diff
@@ -1,5 +1,19 @@
 import { keyMirror } from '@pluralsight/ps-design-system-util'
 
-export const headingSizes = keyMirror('smallCaps', 'medium', 'large', 'xLarge')
+export const capsSizes = keyMirror('small', 'medium', 'large')
+
+export const headingSizes = keyMirror(
+  'xsmall',
+  'small',
+  'medium',
+  'large',
+  'xlarge'
```

Comment:
Why the change from `xLarge` to `xlarge`?

Offhand, I think that it should remain `xLarge`.  This, because:

- It's a JS key, and camelCase is the preferred convention.
- We treat "extra" as a separate word in camelCasing other var names.

For example, in core: (and more)
```
js/breakpoints.ts
6:  xLarge: '1600px'

js/type.ts
8:  fontSizeXXLarge: '36px',
9:  fontSizeXLarge: '30px',
18:  letterSpacingXXLarge: '-0.72px',
19:  letterSpacingXLarge: '-0.54px',

js/layout.ts
7:  spacingXLarge: '48px',
8:  spacingXXLarge: '64px'
```

Reply:
@jaketrent I didn't use camelCase in the size prop string values because it seemed they aren't really two words, but rather a name we conjured to represent a size. 

I'm not sure I understand how the string prop value is a JS key and therefore making camelCase it's preferred convention.

If a component were to have a size prop value of 'xxsmall", what would be the correct camelCasing if 'x' is treated as a separate letter: xXSmall?

Reply:
@sweeting I see, thank you.  It sounds like you re-cased it to `xlarge` because they didn't seem like two words to you.

As to us having "conjured" the name, I feel like that's too-imaginative a description.  `xLarge` stands for "extra large", which seems easily-inferred enough by abbreviating the "x".

The string value becomes a JS key through the keyMirror utility.  It generates an object that will look like:

```
export const sizes = {
  // ...
  xLarge: 'xLarge'
}
```

See [the implementation of keyMirror](https://github.com/pluralsight/design-system/blob/master/packages/util/src/key-mirror.ts)

In the end, yes, it is a preferred convention, whether it's JS or not.  Also, there is ample precedent for this exact convention for this exact word combo.  In addition to the core examples above, here are more:

```
packages/appframe/src/react/index.tsx
161:    const enteringXLarge = prevLargeMedia !== largeMedia && largeMedia
162:    if (enteringXLarge) openSidenav()

packages/avatar/src/vars/index.ts
5:export const sizes = keyMirror('xSmall', 'small', 'medium', 'large', 'xLarge')
34:  xLarge: '160px'

packages/avatar/src/css/index.ts
33:    fontSize: type.fontSizeXXLarge,
42:  [`.psds-avatar--size-${vars.sizes.xLarge}`]: {
43:    width: vars.widths.xLarge,
44:    height: vars.widths.xLarge,

packages/avatar/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap
156:exports[`Storyshots Default xLarge 1`] = `
161:    alt="xLarge avatar"

packages/datawell/src/css/index.ts
34:    fontSize: type.fontSizeXXLarge,

packages/emptystate/src/css/index.ts
28:    padding: `${layout.spacingXLarge} ${layout.spacingLarge}`,

packages/errors/src/css/index.ts
15:    padding: `${layout.spacingXLarge} ${layout.spacingLarge}`,

packages/layout/src/css/page-width-layout.ts
13:      paddingLeft: layout.spacingXLarge,
14:      paddingRight: layout.spacingXLarge

packages/text/src/css/index.ts
71:  [`.psds-text__heading--size-${vars.headingSizes.xLarge}`]: {
72:    fontSize: type.fontSizeXXLarge,
73:    letterSpacing: type.letterSpacingXLarge,

packages/text/src/vars/index.ts
3:export const headingSizes = keyMirror('smallCaps', 'medium', 'large', 'xLarge')

packages/text/src/react/__stories__/index.story.tsx
16:  <div style={{ padding: core.layout.spacingXLarge }}>{storyFn()}</div>

packages/text/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap
207:exports[`Storyshots Heading size: xLarge 1`] = `
218:    xLarge

packages/tooltip/src/react/__stories__/index.story.tsx
10:  <div style={{ padding: core.layout.spacingXLarge }}>{storyFn()}</div>
```

I am open to changing this to `xlarge` on the merits that exist.  I don't see clear ones yet.  And if we did change it, I'd like to change it consistently across the codebase based on understood rules.  My guess is the ROI on that is not great.

And, to the last question, yes, "extra extra small" would be `xXSmall`.

Reply:
I'll change it to align with decision that was made earlier on, but I guess the point I was trying to make is if we abbreviated both 'extra' and 'small' to 'xs' we wouldn't treat both as separate words:

'xXS', 'xS', 's', 'l', 'xL', 'xXL' 

This is probably why numbers are becoming more popular. 🙂

Reply:
Great, thank you.

Yes, this is a harder case -- more of an acronym.  Here, there is less consensus in the community at large.  Happens less, practiced less, and thus is easier to not be consistent.  I've seen and written it in all combost: `xxl`, `XXL` and `xXL`.  `HTML` or `DOM` in code are the most common examples that spring to mind.